<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="mhs_gazebo_ros_control"  params="robot_namespace:='' ">

    <gazebo>
      <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
        <robotNamespace>${robot_namespace}</robotNamespace>
        <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
        <robotParam>robot_description</robotParam>
        <legacyModeNS>true</legacyModeNS>
      </plugin>
    </gazebo>
    
    <gazebo>
      <plugin name="mhs_gazebo_plugin" filename="libmhs_gazebo_plugin.so">
        <robotSimType>m2020_urdf_models/MHSPlugin</robotSimType>
        <bladeVel>251.32</bladeVel>
        <usePoseInput>true</usePoseInput>
        <poseInputTopic>/base_controller/command/pose_cmd</poseInputTopic>
        <velInputTopic>/base_controller/command</velInputTopic>
        <robotNamespace>${robot_namespace}</robotNamespace>
        <xyz>0.0 -2.0 2.5</xyz>
        <noise>0.0001</noise>
      </plugin>
    </gazebo>

  </xacro:macro>  
</robot>
